### PR TITLE
Update actions versions

### PR DESCRIPTION
**Pull request by tinyboxvk**
Fix GitHub Actions Node.js deprecation warnings

`The following actions uses node12 which is deprecated and will be forced to run on node16: actions/checkout@v2, actions/setup-python@v2.`

`Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: actions/checkout@v2, actions/setup-python@v2.`

This PR bumps actions/checkout to v4, and actions/setup-python to v5.

### CODE REVIEW

**Comment by aandergr** [comment]
Pretty nice that you addressed the deprecation warnings in the actions. I think one should also take the opportunity to adjust the versions in the other actions, not only in the Windows EXE build action.

I have also seen that setup-python provides a cache for the dependencies, which could save a few seconds for the build. Would you perhaps like to take the opportunity to implement this as well?

File: .github/workflows/documentation.yml [hunk 1]
```diff
@@ -8,13 +8,14 @@ jobs:
...
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
         with:
           fetch-depth: 0  # needed for building docs
       - name: "Setup Python"
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v5
         with:
           python-version: 3.12
+          cache: 'pip'
```
Comment (aandergr) [inline .github/workflows/documentation.yml:18]:
```suggestion
          cache: 'pipenv'
```

We actually use Pipenv for dependency management, Pip is only used in the Workflows to install Pipenv.

Comment (tinyboxvk) [reply .github/workflows/documentation.yml]: Thanks! I wasn't quite sure which one to use. Does that apply to "packages.yml" as well?

Comment (aandergr) [reply .github/workflows/documentation.yml]: Thanks! Good point, "packages.yml" does acutally not install the Instaloader dependencies, it just installs the dependencies required for the packaging, and does so with Pip. However, the pip cache mechanism seems to be using the requirements.txt file which we do not have, so for packages.yml the cache should better be disabled.

Comment (aandergr) [reply .github/workflows/documentation.yml]: Looks good, thank you very much!

**Pull request merged**